zebra-chain/src/block/header.rs

```diff
@@ -65,3 +65,20 @@ pub struct BlockHeader {
     /// The Equihash solution.
     pub solution: EquihashSolution,
 }
+
+impl BlockHeader {
+    /// Returns true if the header is valid based on its `EquihashSolution`
+    pub fn is_valid(&self) -> bool {
```

Comment:
Issue: `BlockHeader.is_valid()` is easily confused with validation for all the fields in the header.

Suggestion: Let's call it `is_equihash_solution_valid()`, or something similar.